### PR TITLE
proposal: encapsulate multitype headers and clean

### PR DESCRIPTION
**Pull request by wrossmorrow**

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -29,12 +30,15 @@ type PhaseResponse struct {
...
-	Scheme      string
-	Authority   string
-	Method      string
-	Path        string
-	RequestId   string
+	Scheme    string
+	Authority string
+	Method    string
+	Path      string
+	RequestId string
```
Comment (izturn) [inline context.go]: RequestID

File: context.go [hunk 2]
```diff
@@ -29,12 +30,15 @@ type PhaseResponse struct {
...
+
 	Headers     map[string][]string
+	ByteHeaders map[string][]byte
```
Comment (izturn) [inline context.go]: `RawHeaders` to match envoy's `raw_values`?

File: context.go [hunk 3]
```diff
@@ -43,41 +47,53 @@ type RequestContext struct {
...
 		default:
-			rc.Headers[h.Key] = strings.Split(h.Value, ",")
+			rc.Headers[k] = v
 		}
 	}
 
 	return nil
 }
 
+func (rc *RequestContext) GetAllHeaders() AllHeaders {
```
Comment (izturn) [inline context.go]: `AllHeaders`?, the`GetXX` is not very go style

Comment (wrossmorrow) [reply context.go]: Updated

File: context.go (hunk 2 above)
Comment (wrossmorrow) [reply context.go]: Updated

File: context.go (hunk 1 above)
Comment (wrossmorrow) [reply context.go]: Updated

**Comment by izturn** [comment]
@wrossmorrow hey, i'd like to use it in production, let's merge it into main first?

**Comment by izturn** [comment]
ping @wrossmorrow

**Comment by wrossmorrow** [comment]
Sorry, somehow totally missed your replies

**Comment by wrossmorrow** [comment]
Do you have any other comments? Would you like to discuss your use case?

**Comment by izturn** [comment]
my usecase is very simple, just  use the package to make some demos like the all under `examples` , at current.

**Comment by wrossmorrow** [comment]
If you approve I can merge (or just override of course)

**Comment by wrossmorrow** [comment]
Why don't I just pull the trigger; ping me if there's another issue.

**Pull request merged**